WiinUPro/Windows/DevicePrefsWindow.xaml.cs

```diff
@@ -116,15 +166,22 @@ private void cancelBtn_Click(object sender, RoutedEventArgs e)
 
         private void btnDefaultProfile_Click(object sender, RoutedEventArgs e)
         {
-            Microsoft.Win32.OpenFileDialog dialog = new Microsoft.Win32.OpenFileDialog();
-            dialog.DefaultExt = ".wup";
-            dialog.Filter = App.PROFILE_FILTER;
+            FindProfile(defaultProfile);
+        }
 
-            bool? didFind = dialog.ShowDialog();
+        private void btnExtProfile_Click(object sender, RoutedEventArgs e)
+        {
+            FindProfile(extProfile);
+            _modifiedPrefs.extensionProfiles[comboExtProfile.SelectedIndex] = extProfile.Text;
+        }
 
-            if (didFind == true && dialog.CheckFileExists)
+        private void comboExtProfSelect_SelectionChanged(object sender, SelectionChangedEventArgs e)
+        {
+            if (selectionChanged)

```

Comment:
This is because this fires when the component is made correct?

Reply:
Yes, at line 31 and sometimes also at line 97.